tests/test_exceptions.py

```diff
@@ -333,3 +345,71 @@ def test_match_processing(
     with pytest.raises(MatchProcessing) as exc:
         cloud_reco_client.query(image=high_quality_image)
     assert exc.value.response.status_code == HTTPStatus.INTERNAL_SERVER_ERROR
+
+
+def test_vwsexception_inheritance() -> None:
+    """
+    VWS-related exceptions should inherit from VWSException.
+    """
+    subclasses = [
+        AuthenticationFailure,
+        BadImage,
+        DateRangeError,
+        Fail,
+        ImageTooLarge,
+        MetadataTooLarge,
+        ProjectInactive,
+        ProjectHasNoAPIAccess,
+        ProjectSuspended,
+        RequestQuotaReached,
+        RequestTimeTooSkewed,
+        TargetNameExist,
+        TargetQuotaReached,
+        TargetStatusNotSuccess,
+        TargetStatusProcessing,
+        UnknownTarget,
+        UnknownVWSErrorPossiblyBadName,
+    ]
+    for subclass in subclasses:
+        assert issubclass(subclass, VWSException)
+
+
+def test_cloudrecoexception_inheritance() -> None:
+    """
+    CloudRecoService-specific exceptions should inherit from
+    CloudRecoException.
+    """
+    subclasses = [
+        MatchProcessing,
+        MaxNumResultsOutOfRange,
+    ]
+    for subclass in subclasses:
+        assert issubclass(subclass, CloudRecoException)
+
+
+def test_others_inheritance() -> None:
+    """
+    Make sure other exceptions are inherited from their expected super-classes.
+    """
+    assert issubclass(
+        ConnectionErrorPossiblyImageTooLarge,
+        requests.ConnectionError,
+    )
+    assert issubclass(TargetProcessingTimeout, Exception)
+
+
+def test_base_exceptions_have_response_property_and_text_str() -> None:
+    """
+    A VWSException or CloudRecoException should have a response property
+    and string representation with the text property of the response.
+    """
+    base_classes: List[Union[Type[CloudRecoException], Type[VWSException]]] = [
+        CloudRecoException,
+        VWSException,
+    ]
+    for base in base_classes:
+        response = requests.Response()
+        setattr(response, '._content', bytes(f'Test{base.__name__}', 'ascii'))
```

Comment:
Thanks, FYI I missed something here, it should be `'_content'` instead of `'._content'`.